mesonbuild/interpreter/interpreter.py

```diff
@@ -316,6 +316,7 @@ def __init__(
         self.build_holder_map()
         self.user_defined_options = user_defined_options
         self.compilers: PerMachine[T.Dict[str, 'compilers.Compiler']] = PerMachine({}, {})
+        self.rsp_files: T.Dict[str, int] = {}  # number of rsp_file per subdir, in order to generate unique names
```

Comment:
You can use an automatic counter: `self.rsp_files: T.Dict[str, itertools.count] = collections.defaultdict(itertools.count)`

Reply:
interresting. Never did that before...

Reply:
I learned that from that code: https://github.com/mesonbuild/meson/blob/master/mesonbuild/interpreter/compiler.py#L778. It does exactly the same thing.